torch/utils/data/graph_settings.py

```diff
@@ -42,7 +46,7 @@ def apply_sharding(datapipe: DataPipe, num_of_instances: int, instance_id: int)
                     if already_applied_to is not None:
                         raise RuntimeError('This implementation of sharding can be only applied once per instance of DataPipeline.',
                                            'Already applied to', already_applied_to, 'while trying to apply to', pipe)
-                    pipe.apply_sharding(num_of_instances, instance_id)
+                    pipe.apply_sharding(num_of_instances, instance_id, sharding_group=sharding_group)
```

Comment:
noob question: does `is_shardable` and `apply_sharding` only exists in `ShardingFilterIterDataPipe` ? 

Also if there is no `ShardingFilterIterDataPipe`, look like no sharding will happen? (shall we error in that case? ) :) 

cc @ejguan